clients/java/client/src/it/java/org/camunda/bpm/client/client/ClientIT.java

```diff
@@ -252,6 +250,92 @@ public void shouldThrowExceptionDueToMaxTasksNotGreaterThanZero() {
     }
   }
 
+
+  @Test
+  public void shouldUseExternalTaskClientListener() {
+
+    ExternalTaskClient client = null;
+    List<String> states = new LinkedList<>();
+
+    try {
+    // given
+    engineRule.startProcessInstance(processDefinition.getId());
+
+    ExternalTaskClientBuilder clientBuilder = ExternalTaskClient.create()
+            .externalTaskClientListener(new DefaultClientInteractionListener() {
+
+              @Override
+              public void onFetchAndLock(List<TopicRequestDto> topics) {
+                System.out.println("state: onFetchAndLock");
+                states.add("onFetchAndLock");
+              }
+
+              @Override
+              public void fetchAndLockDone(List<TopicRequestDto> topics, List<ExternalTask> externalTasks) {
+                System.out.println("state: fetchAndLockDone");
+                states.add("fetchAndLockDone");
+              }
+
+              @Override
+              public void fetchAndLockFail(List<TopicRequestDto> topics, EngineClientException e) {
+                System.out.println("state: fetchAndLockFail");
+                states.add("fetchAndLockFail");
+              }
+
+              @Override
+              public void onSetVariable(String processInstanceId, Map<String, Object> variables) {
+                System.out.println("state: onSetVariable");
+                states.add("onSetVariable");
+              }
+
+              @Override
+              public void setVariableDone(String processInstanceId, Map<String, Object> variables) {
+                System.out.println("state: setVariableDone");
+                states.add("setVariableDone");
+              }
+
+              @Override
+              public void setVariableFail(String processInstanceId, Map<String, Object> variables, EngineClientException e) {
+                System.out.println("state: setvariableFail");
+                states.add("setvariableFail");
+              }
+
+            })
+            .baseUrl(BASE_URL);
+      // then
+      thrown.expect(ExternalTaskClientException.class);
+
+      // when
+      client = clientBuilder.build();
+      client.subscribe(EXTERNAL_TASK_TOPIC_FOO)
+              .handler(handler)
+              .open();
+
+      Object[] assertStates = {
+              "onFetchAndLock", "fetchAndLockDone",
+              "onSetVariable", "setVariableDone"
+      };
+
+      while (states.size() < assertStates.length)
+        try {
+          Thread.sleep(10);
+        } catch (InterruptedException ignored) {
+        }
+
+      List<String> firstStates = new ArrayList<>(assertStates.length);
+
+      for (int i = 0; i < assertStates.length; i++) firstStates.add(states.get(i));
```

Comment:
hello again @tmetzke

I have copied it just for better assertion as `assertArrayEquals`  :slightly_smiling_face:  , because we have a `LinkedList (states)` that could extended according to our used listeners.

I think there is not any assertion concept for LinkedList, Let me know if you have any better suggestion :slightly_smiling_face: 

Thank you in advance

Reply:
I would encourage you to have a look at AssertJ instead of JUnit's Assert class. This is what we use in this class here anyways and can also do something like `assertThat(states).containsExactlyInAnyOrderElementsOf(assertStates)`. Play around with that and see if something matches what you would like to assert here.